extensions/extension.go

```diff
@@ -115,3 +141,42 @@ func sprintf(args []object.Object) object.Object {
 	res := fmt.Sprintf(args[0].(object.String).Value, object.Unwrap(args[1:])...)
 	return object.String{Value: res}
 }
+
+func convertMap(m map[any]any) map[string]any {
+	result := make(map[string]any, len(m))
+	for key, value := range m {
+		if valueMap, ok := value.(map[any]any); ok {
+			value = convertMap(valueMap)
+		}
+		result[fmt.Sprint(key)] = value
```

Comment:
Maybe with the type because 1 and 1. and true may collide no?

Like 1, 1., and "1"

Maybe it's OK, but I'm raising the point

Reply:
I'm dropping this code in favor of handcoded serialization for all objects (commit incoming, was just debugging why I still saw some encoding/json in GoSizeAnalyzer and it turned out it was from fortio.org/sets, I just made a new release that has the `no_json` tag)

But to answer your question, "1" and 1 and "1." do collide (but not true, true is/was "true")

I could prefix the key with the type, I'm actually doing that for a few types like error in the upcoming commit

Reply:
I'm sorry I edited my message after checking, but forgot one part.

Yes `true` doesn't collide with 1, but it does with `"true" `

So yes, also using the type could avoid collision.

Reply:
The way you handled is elegant

Reply:
Thanks! of note that yes the translation to json and then back isn't identity (at least not yet, unless I would implement custom deserialization) the current goal is more so people can produce json out of their grol data structures